crates/core/app/src/action_handler/actions/submit.rs

```diff
@@ -203,6 +223,30 @@ impl ActionHandler for ProposalSubmit {
             ProposalPayload::UpgradePlan { .. } => {
                 // TODO(erwan): no stateful checks for upgrade plan.
             }
+            ProposalPayload::UnplannedIbcUpgrade { .. } => {
+                // TODO: which checks happen here?
+                todo!();
```

Comment:
I suggest we ship this PR with Freeze/Unfreeze and leave out trying to handle arbitrary client modifications for now, and remove the `todo!()`. The reason for this is that we do not have a great idea yet of the both the scope of changes that need to happen there.

Reply:
Agree with this